File: audiobook/src/main/java/de/ph1b/audiobook/dialog/DonationDialogFragment.java
package de.ph1b.audiobook.dialog;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

import com.afollestad.materialdialogs.MaterialDialog;

import de.ph1b.audiobook.R;


public class DonationDialogFragment extends DialogFragment {
    public static final String TAG = DonationDialogFragment.class.getSimpleName();

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final MaterialDialog.ListCallback donationListCallback = (materialDialog, view, i, charSequence) -> {
            String item;
            switch (i) {
                case 0:
                    item = "1donation";
                    break;
                case 1:
                    item = "2donation";
                    break;
                case 2:
                    item = "3donation";
                    break;
                case 3:
                    item = "5donation";
                    break;
                case 4:
                    item = "10donation";
                    break;
                case 5:
                    item = "20donation";
                    break;
                default:
                    throw new AssertionError("There are only 4 items");
            }
            ((OnDonationClickedListener) getActivity()).onDonationClicked(item);
        };

        return new MaterialDialog.Builder(getActivity())
                .title(R.string.pref_support_donation)
                .items(R.array.pref_support_money)
                .itemsCallback(donationListCallback)
                .build();
    }

    public interface OnDonationClickedListener {
        void onDonationClicked(String item);
    }
}

File: audiobook/src/main/java/de/ph1b/audiobook/dialog/DonationDialogFragment.kt
package de.ph1b.audiobook.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.support.v4.app.DialogFragment

import com.afollestad.materialdialogs.MaterialDialog

import de.ph1b.audiobook.R


class DonationDialogFragment : DialogFragment() {

    private lateinit var callback: OnDonationClickedListener

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val donationListCallback = MaterialDialog.ListCallback { materialDialog, view, i, charSequence ->
            val item: String
            when (i) {
                0 -> item = "1donation"
                1 -> item = "2donation"
                2 -> item = "3donation"
                3 -> item = "5donation"
                4 -> item = "10donation"
                5 -> item = "20donation"
                else -> throw AssertionError("There are only 4 items")
            }
            callback.onDonationClicked(item)
        }

        return MaterialDialog.Builder(context)
                .title(R.string.pref_support_donation)
                .items(R.array.pref_support_money)
                .itemsCallback(donationListCallback)
                .build()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)

        callback = context as OnDonationClickedListener
    }

    interface OnDonationClickedListener {
        fun onDonationClicked(item: String)
    }

    companion object {
        val TAG = DonationDialogFragment::class.java.simpleName
    }
}
